docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,358 @@
+  --pin <sha256:...> \
+  --cert <path> \
+  --password <pw> \
+  --password-prompt \
+  --cosign \
+  --cosign-upload \
+  --cosign-skip-tlog \
+  --cosign-annotation <key=value> \
+  --cosign-annotation <key=value> \
+  --cosign-identity-token <token-or-path>
+```
+
+- `<ref>`: descriptor reference (file path or repository reference).
+- `--recurse`: calculate digests for referenced component versions.
+- `--force`: overwrite existing digest fields.
+- `--sig <slot>`: logical signature slot/name (e.g., `mysig@1234`).
+- `--pin <sha256:...>`: expected component-version digest; fail on mismatch.
+- `--cert <path>`: certificate/key material for direct signing (PEM/PKCS#8/PKCS#12).
+- `--password <pw>` / `--password-prompt`: decrypt key material if protected.
+- `--cosign`: also produce an OCI Cosign signature for the same canonical payload.
+- `--cosign-upload`: if `<ref>` resolves to an OCI subject, attach/upload the Cosign signature to the registry; otherwise, store the Cosign bundle only.
+- `--cosign-skip-tlog`: disable Rekor upload.
+- `--cosign-annotation`: repeatable `key=value` annotations for the Cosign signature.
+- `--cosign-identity-token <token-or-path>`: pass an OIDC ID token directly (alternatively, use `SIGSTORE_ID_TOKEN` env).
+
+### Normalization
+
+- Canonicalization for component-version digest and signature payload is **JCS (RFC 8785)** via the existing `jcs` package.
+- No CLI flag to select a different normalization.
+
+---
+
+## Behavior
+
+### Digest calculation (`ocm add digests cv`)
+
+- Computes and **embeds** digests:
+  - **Resources**: content digests.
+  - **Component references**: digest of the referenced component descriptor.
+  - **Component-version**: digest over the canonicalized descriptor.
+- May download artifacts and referenced descriptors as needed.
+- Writes the updated descriptor back to `<ref>`.
+
+### Signing (`ocm sign cv`)
+
+- Loads the descriptor from `<ref>`.
+- Canonicalizes the descriptor using JCS.
+- Computes the component-version digest from canonical bytes.
+- If `--pin` is provided, compares and fails on mismatch.
+- Performs direct signing using `--cert` (or a default signer if available).
+- If `--cosign` is set:
+  - Obtain an OIDC ID token in this order:
+    1. Use `--cosign-identity-token` if provided (token string or `@path` to file).
+    2. Else use `SIGSTORE_ID_TOKEN` environment variable if present.
+    3. Else, if interactive TTY is available, run an **interactive loopback** browser flow to retrieve the token.
+    4. Else, run the **device flow** to retrieve the token.
+  - Produce a Cosign signature over the **same canonical bytes**.
+  - When `--cosign-upload` is present and `<ref>` maps to an OCI subject (`<name>@<digest>`), attach the Cosign signature to the registry using Cosign conventions.
+  - Record the **Cosign bundle** (signature, certs, and optional Rekor inclusion) in the envelope for offline verification.
+- Append the signature envelope under `.signatures[]`.
```

Comment:
the signature envelope is not arbitrary and you have to adhere to the OCM specification